app/components/build/AvatarWithUnknownEmailPrompt.js

```diff
@@ -0,0 +1,321 @@
+
+    Relay.Store.commitUpdate(mutation, { onSuccess: this.handleEmailAddedSuccess, onFailure: this.handleMutationFailure });
+  };
+
+  handleResendVerificationClick = () => {
+    this.setState({ isSendingVerification: true });
+
+    const mutation = new EmailResendVerificationMutation({ email: this.getUserEmail(this.props.build.createdBy.email) });
+
+    Relay.Store.commitUpdate(mutation, { onSuccess: this.handleVerificationResentSuccess, onFailure: this.handleMutationFailure });
+  };
+
+  handleDismissedSucess = () => {
+    this.setState({ isDismissing: false });
+  };
+
+  handleEmailAddedSuccess = () => {
+    this.setState({ isAddingEmail: false, hasSentSomething: true });
+  };
+
+  handleVerificationResentSuccess = () => {
+    this.setState({ isSendingVerification: false, hasSentSomething: true });
+  };
+
+  handleLocalDismissClick = () => {
+    this.setState({ hasBeenDismissed: true });
+  }
+
+  getMessages(loading) {
+    // There won't be an email address if this build was created by a
+    // registered user or if this build just has no owner (perhaps it was
+    // created by Buildkite)
+    if (!this.props.build.createdBy.email) {
+      return {};
+    }
+
+    // If we haven't decided to send a query for this yet, don't show anything!
+    if (!this.props.relay.variables.isTryingToPrompt) {
+      return {};
+    }
+
+    let message;
+    let buttons;
+
+    const userEmail = this.getUserEmail(this.props.build.createdBy.email);
+
+    if (!this.state.isAddingEmail && userEmail) {
+      if (userEmail.verified) {
+        return {};
+      }
+
+      message = (
+        <div>
+          <h1 className="h5 m0 mb1 bold">Email verification needed</h1>
+          <p>We’ve sent a verification email to <strong className="semi-bold">{this.props.build.createdBy.email}</strong>. Click the link in that email to finish adding it to your account.</p>
+          <p className="dark-gray mt0 dark-gray m0 h7">You can resend the verification email or remove this email address in your <a className="semi-bold lime hover-lime hover-underline" href="/user/emails">Personal Email Settings</a></p>
+        </div>
+      );
+
+      if (this.state.isSendingVerification || !this.state.hasSentSomething) {
+        message = (
+          <div>
+            <h1 className="h5 m0 mb1 bold">Email verification needed</h1>
+            <p>Please click the link in the verification email we sent to <strong className="semi-bold">{this.props.build.createdBy.email}</strong>.</p>
+            <p className="dark-gray mt0 dark-gray m0 h7">You can resend the verification email or remove this email address in your <a className="semi-bold lime hover-lime hover-underline" href="/user/emails">Personal Email Settings</a></p>
+          </div>
+        );
+      }
+    } else {
+      // Otherwise, we've got an unknown (to Buildkite) email address on our hands!
```

Comment:
For some reason I read this and then said this in my head:

![50538596](https://cloud.githubusercontent.com/assets/25882/22391032/cc8d8088-e542-11e6-972b-21d7ccee0ed9.jpg)